### PR TITLE
Fix the tests and add CI to run them

### PR DESCRIPTION
**Pull request by jaimesouza**
**Please provide a description of the purpose of this pull request, as well as its
motivation and context.**

This PR fixes the tests that were failing.
It also adds a GitHub workflow (CI) to the tests on `push` and `pull request` events to the master branch.

**How was the code tested?**

It was tested locally with both Centos 7 and Ubuntu focal LXD-deployed charms.

Also, the CI itself runs the tests.

Final checklist
- [x] I am the author of these changes, or I have the rights to submit them.
- [ ] I added the relevant changed to the README and/or documentation.
- [x] I self reviewed my own code.
- [ ] I updated the `CHANGELOG` according to the [contributing
  guide](https://omnivector-solutions.github.io/osd-documentation/master/contributing.html#changelog)

### CODE REVIEW

File: .github/workflows/run-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Unit Tests
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
```
Comment (fschuch) [inline .github/workflows/run-tests.yaml]: What do you think of running the tests for every PR?

```suggestion

```

**Review by fschuch — CHANGES_REQUESTED** [review]
Well done!

File: .github/workflows/run-tests.yaml (hunk 1 above)
Comment (jaimesouza) [reply .github/workflows/run-tests.yaml]: I think it's a great ideia. Thanks!

**Review by dusktreader — APPROVED** [review]
Nice work!

File: Makefile [hunk 2]
```diff
@@ -66,9 +66,16 @@ slurmrestd: version ## pack slurmrestd
...
+.PHONY: tests-centos
+tests-centos: clean charms ## Run bats tests on centos
+	./run_tests centos
+
+.PHONY: tests-focal
+tests-focal: clean charms ## Run bats tests on ubuntu focal
+	./run_tests focal
+
+.PHONY: tests-all
+tests-all: tests-centos tests-focal ## Run all bats tests
```
Comment (heitorPB) [inline Makefile]:
```suggestion
.PHONY: tests
tests: tests-centos tests-focal ## Run bats tests
```

File: .github/workflows/run-tests.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+name: Unit Tests
```
Comment (heitorPB) [inline .github/workflows/run-tests.yaml]:
```suggestion
name: Integration Tests
```

**Comment by jaimesouza** [comment]
Hey @heitorPB, 

The test below fails randomly. I mean, it usually fails, but if I run the action again, It successfully passes.

Can you think of a reason for that? 

```
not ok 13 test we can set a default partition
# (from function `assert_output' in file tests/../node_modules/bats-assert/src/assert.bash, line 247,
#  in test file tests/10-slurm-partition.bats, line 32)
#   `assert_output --partial "$partition* "' failed
#
# -- output does not contain substring --
# substring (1 lines):
#   Default-test-partition*
# output (3 lines):
#   PARTITION              AVAIL  TIMELIMIT  NODES  STATE NODELIST
#   Default-test-partition    up   infinite      1   idle juju-a0a8b2-2
#   Default-test-partition    up   infinite      1   down juju-a0a8b2-5
# --
```

**Pull request merged**